File: BackEnd/src/app/infraestructure/mappers/VideoMapper.ts
import { Video } from "../../domain/entities/Video";
import { VideoRecord } from "../datasorces/intern/models/VideoRecord";

export class VideoMapper {
  static toDomain(record: VideoRecord): Video {
    return {
      id: record.id,
      nom: record.nom,
      descripcio: record.descripcio,
      duration: record.duration,
      thumbnail: record.thumbnail,
      videoUrl: record.videoUrl
    };
  }

  static toRecord(domain: Video): VideoRecord {
    return {
      id: domain.id,
      nom: domain.nom,
      descripcio: domain.descripcio,
      duration: domain.duration,
      thumbnail: domain.thumbnail,
      videoUrl: domain.videoUrl 
    };
  }
}

File: FrontEnd/lib/features/data/mappers/video_mapper.dart
import '../../domain/entities/video.dart';

class VideoMapper {
  static Video fromJson(Map<String, dynamic> json) => Video(
        id: json['id'],
        nom: json['nom'] ?? '',
        descripcio: json['descripcio'] ?? '',
        duration: json['duration'] ?? 0,
        thumbnail: json['thumbnail'] ?? '',
        url: json['url'],
      );
}
